### PR TITLE
server: add basic Server metrics

### PR DESCRIPTION
**Pull request by creachadair**
- counter_api_calls: total calls to each API method
 - counter_api_bad_request: invalid requests (by method)
 - counter_api_forbidden: forbidden (by method)
 - counter_api_internal_error: other errors (by method)

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -43,6 +43,12 @@ type Config struct {
 type Server struct {
 	db    *db.DB
 	whois func(context.Context, string) (*apitype.WhoIsResponse, error)
+
+	// Metrics
+	countCalls             expvar.Map // :: method name → count
```
Comment (danderson) [inline server/server.go:48]: use metrics.LabelMap instead, to produce prometheus friendly metric names

File: server/server.go [hunk 2]
```diff
@@ -65,6 +71,17 @@ func New(cfg Config) (*Server, error) {
 	return ret, nil
 }
 
+// Metrics returns a collection of metrics for s. THe caller is responsible for
+// publishing the result to the metrics exporter.
+func (s *Server) Metrics() expvar.Var {
+	m := new(expvar.Map)
```
Comment (danderson) [inline server/server.go:77]: this one can remain a regular map, its a container of unrelated metrics.

File: server/server.go (hunk 1 above)
Comment (creachadair) [reply server/server.go]: Done! PTAL.

File: server/server.go (hunk 2 above)
Comment (creachadair) [reply server/server.go]: I turned it into a metrics.Set

**Review by danderson — APPROVED** [review]

**Pull request merged**